scripts/exrrfs_lbc.sh

```diff
@@ -26,9 +26,9 @@ fi
 #
 # find start and end time
 #
-fhr_chunk=$(( (10#${LENGTH}/10#${INTERVAL} + 1)/10#${GROUP_NUM}*10#${INTERVAL} ))
+fhr_chunk=$(( (10#${LENGTH}/10#${INTERVAL} + 1)/10#${GROUP_TOTAL_NUM}*10#${INTERVAL} ))
 fhr_begin=$((10#${OFFSET} + (10#${GROUP_INDEX} - 1 )*10#${fhr_chunk} ))
-if [[ 10#${GROUP_INDEX} -eq 10#${GROUP_NUM} ]]; then
+if [[ 10#${GROUP_INDEX} -eq 10#${GROUP_TOTAL_NUM} ]]; then
```

Comment:
This is not required by NCO, but a good practice based on lessons learned from previous errors.  

In Bash, use the double parentheses `(( ))` instead of `[[ ]]` for arithmetic operations and comparisons. The former is more intuitive and less error-prone. For example, use `if (( 10#$num1 > 10#$num2 )); then`, avoid `if [[ "${num1}" -gt "${num2}" ]]; then`. 

Also `[[` does not support enforcing base10.

Reply:
To clarify, use `((` for arithmetic comparisons while `[[` for all other comparisons.

Reply:
Sorry, I forgot that neither `[[` nor `((` supports enforcing base10.  
Only `$((....))` can enforce base10.

For `if [[ 10#${GROUP_INDEX} -eq 10#${GROUP_TOTAL_NUM} ]]; then` , it actually runs as follows
```
GROUP_INDEX=2
GROUP_TOTAL_NUM=2
[[ 10#2 -eq 10#2 ]] 
```
not `[[ 2 -eq 2 ]]`.

For `if (( 10#${GROUP_INDEX} == 10#${GROUP_TOTAL_NUM} )); then`, it shows an error like this:
```
test.sh: line 7: ((: 10#002 -eq 10#2 : syntax error in expression (error token is "10#2 ")
```

Reply:
If we want to enforce base10, we will need to do 
`if (( $((10#${GROUP_INDEX})) == $((10#${GROUP_TOTAL_NUM})) )); then`
This is a solution similar to this PR #443 

Here is a test example:
```
#!/bin/bash
GROUP_INDEX=02                                                                                                                                                              
GROUP_TOTAL_NUM=002
if (( $((10#${GROUP_INDEX})) == $((10#${GROUP_TOTAL_NUM})) )); then
  echo "yes"
else
  echo "no"
fi
```

Reply:
Or we can simply use:
`if (( ${GROUP_INDEX} == ${GROUP_TOTAL_NUM} )); then`
But this will require GROUP_INDEX and GROUP_TOTAL_NUM to be set correctly (without any leading zeros, etc).

Here is a test example:
```
#!/bin/bash
GROUP_INDEX=9
GROUP_TOTAL_NUM=9
if (( ${GROUP_INDEX} == ${GROUP_TOTAL_NUM} )); then
  echo "yes"
else                                                                                                                                                                        
  echo "no"
fi
```
If we accidentally change "GROUP_INDEX" to 009, it will not be robust as the above base10 example and will complain as follows:
`test.sh: line 4: ((: 009: value too great for base (error token is "009")`